src/hooks/useTokenDetails.ts

```diff
@@ -1,20 +1,25 @@
 import { useState, useContext, useEffect } from "react";
-import { TokenBalance } from "types";
+import { TokenDetails } from "types";
 
-import { Web3Context } from "components/Web3Provider";
+import {
+  ContractInteractionContext,
+  ContractInteractionContextProps,
+} from "components/context/ContractInteractionProvider";
 
 interface Return {
-  tokenDetails: TokenBalance | null;
+  tokenDetails?: TokenDetails;
   isLoading: boolean;
   error: string;
 }
 
 export function useTokenDetails(token?: string): Return {
-  const [tokenDetails, setTokenDetails] = useState<TokenBalance | null>(null);
+  const [tokenDetails, setTokenDetails] = useState<TokenDetails>(null);
```

Comment:
I think this won't work on strict mode, as the `null` value is being set while it now expects a `TokenDetails`